### PR TITLE
PostGIS E2E test

### PR DESCRIPTION
**Pull request by tjmoore4**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**

 
 - [ ] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [x] Testing enhancement
 - [ ] Other

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
This update adds a Kuttl E2E test for PostGIS enabled clusters.
The test is generated similar to the other E2E tests, but is stored
in the 'e2e-other' directory to better facilitate testing the two
PostGIS version images per PostgreSQL version that are created.

This test checks the postgis_version and postgis_full_version values,
ensures that PostGIS schemas exist and tests the 'ST_MakePoint'
PostGIS function. If any check fails, the Job will error.

Also, the GSSAPI test is updated to use environment substitution
instead of a fixed Postgres version and PSQL image.

**Other Information**:
Issue: [sc-13960]

### CODE REVIEW

File: testing/kuttl/kuttl-test.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ apiVersion: kuttl.dev/v1beta1
 kind: TestSuite
 testDirs:
 - testing/kuttl/e2e-generated/
+- testing/kuttl/e2e-generated-other/
```
Comment (jmckulk) [inline testing/kuttl/kuttl-test.yaml:5]: I don't think we would want to run this as part of the standard `make check-kuttl` command. I think having it defined here would try to run it

File: Makefile [hunk 2]
```diff
@@ -219,17 +219,31 @@ check-kuttl:
 
 .PHONY: generate-kuttl
 generate-kuttl: export KUTTL_PG_VERSION ?= 14
-generate-kuttl: export KUTTL_PSQL_IMAGE ?= registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.1-0
+generate-kuttl: export KUTTL_PSQL_IMAGE ?= registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.2-0
 generate-kuttl:
 	[ ! -d testing/kuttl/e2e-generated ] || rm -r testing/kuttl/e2e-generated
 	bash -ceu ' \
-	render() { envsubst '"'"'$$KUTTL_PG_VERSION $$KUTTL_PSQL_IMAGE'"'"'; }; \
+	render() { envsubst '"'"'$$KUTTL_PG_VERSION $$KUTTL_POSTGIS_VERSION $$KUTTL_PSQL_IMAGE'"'"'; }; \
```
Comment (jmckulk) [inline Makefile:226]: I think we can remove the `$$KUTTL_POSTGIS_VERSION` from this make target :wrench:

File: Makefile [hunk 3]
```diff
@@ -219,17 +219,31 @@ check-kuttl:
...
 	while [ $$# -gt 0 ]; do \
 		source="$${1}" target="$${1/e2e/e2e-generated}"; \
 		mkdir -p "$${target%/*}"; render < "$${source}" > "$${target}"; \
 		shift; \
 	done' - $(wildcard testing/kuttl/e2e/*/*.yaml)
 
+.PHONY: generate-kuttl-other
+generate-kuttl-other: export KUTTL_PG_VERSION ?= 14
```
Comment (jmckulk) [inline Makefile:234]: I don't like to see this duplicated but it's fine. I do wonder if we could have something like this :recycle:
```
kuttl-envs: export BLAH
kuttl-envs: export BLAB

generate-kuttl:  kuttl-envs
<stuff>

generate-kuttl-other: kuttl-envs
generate-kuttl-other: export BLAH-other
<other stuff>
```

**Review by jmckulk — COMMENTED** [review]
Couple small things

File: Makefile (hunk 2 above)
Comment (tjmoore4) [reply Makefile:226]: Yeah, I missed that one. Thanks, updated.

File: testing/kuttl/kuttl-test.yaml (hunk 1 above)
Comment (tjmoore4) [reply testing/kuttl/kuttl-test.yaml:5]: :thinking: I can see that being an issue. I'll add a note in the README with an alternate way to run the test.

File: Makefile (hunk 3 above)
Comment (tjmoore4) [reply Makefile:234]: I tried something similar, but unfortunately couldn't get the syntax correct. Maybe there's an another way to do this? For now, I'll leave this as is.

**Review by jmckulk — APPROVED** [review]
LGTM

File: Makefile (hunk 3 above)
Comment (cbandy) [reply Makefile:234]: 🔧 I recommend combining these so we have only the `generate-kuttl` target.

```diff
diff --git Makefile Makefile
index 41cb2804d..0de84393b 100644
--- Makefile
+++ Makefile
@@ -219,22 +219,10 @@ check-kuttl:
 
 .PHONY: generate-kuttl
 generate-kuttl: export KUTTL_PG_VERSION ?= 14
+generate-kuttl: export KUTTL_POSTGIS_VERSION ?= 3.1
 generate-kuttl: export KUTTL_PSQL_IMAGE ?= registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.2-0
 generate-kuttl:
 	[ ! -d testing/kuttl/e2e-generated ] || rm -r testing/kuttl/e2e-generated
-	bash -ceu ' \
-	render() { envsubst '"'"'$$KUTTL_PG_VERSION $$KUTTL_PSQL_IMAGE'"'"'; }; \
-	while [ $$# -gt 0 ]; do \
-		source="$${1}" target="$${1/e2e/e2e-generated}"; \
-		mkdir -p "$${target%/*}"; render < "$${source}" > "$${target}"; \
-		shift; \
-	done' - $(wildcard testing/kuttl/e2e/*/*.yaml)
-
-.PHONY: generate-kuttl-other
-generate-kuttl-other: export KUTTL_PG_VERSION ?= 14
-generate-kuttl-other: export KUTTL_POSTGIS_VERSION ?= 3.1
-generate-kuttl-other: export KUTTL_PSQL_IMAGE ?= registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.2-0
-generate-kuttl-other:
 	[ ! -d testing/kuttl/e2e-generated-other ] || rm -r testing/kuttl/e2e-generated-other
 	bash -ceu ' \
 	render() { envsubst '"'"'$$KUTTL_PG_VERSION $$KUTTL_POSTGIS_VERSION $$KUTTL_PSQL_IMAGE'"'"'; }; \
@@ -242,7 +230,7 @@ generate-kuttl-other:
 		source="$${1}" target="$${1/e2e/e2e-generated}"; \
 		mkdir -p "$${target%/*}"; render < "$${source}" > "$${target}"; \
 		shift; \
-	done' - $(wildcard testing/kuttl/e2e-other/*/*.yaml)
+	done' - $(wildcard testing/kuttl/e2e/*/*.yaml) $(wildcard testing/kuttl/e2e-other/*/*.yaml)
 
 .PHONY: check-generate
 check-generate: generate-crd generate-deepcopy generate-rbac
```

File: Makefile [hunk 4]
```diff
@@ -240,6 +254,7 @@ clean: clean-deprecated
 	rm -f bin/postgres-operator
 	rm -f config/rbac/role.yaml
 	[ ! -d testing/kuttl/generated ] || rm -r testing/kuttl/generated
+	[ ! -d testing/kuttl/generated-other ] || rm -r testing/kuttl/generated-other
```
Comment (cbandy) [inline Makefile:257]: ❓ Should these two lines be `testing/kuttl/e2e-generated` and `testing/kuttl/e2e-generated-other`?

Comment (tjmoore4) [reply Makefile:257]: Yes, that looks like an oversight. Thanks  for the catch, updated.

File: Makefile (hunk 3 above)
Comment (tjmoore4) [reply Makefile:234]: :+1: That works for me, the generation is pretty fast regardless.

**Review by jmckulk — APPROVED** [review]

**Pull request merged**